src/client/components/Home.jsx

```diff
@@ -142,7 +105,7 @@ export default class Home extends Component {
   updateValue(fieldname, value) {
     this.setState((currentState) => {
       const diaryPage = Object.assign(currentState.diaryPage, { [fieldname]: value });
-      window.history.pushState(null, '', Home.getUrl({ diaryPage }));
+      window.history.pushState(null, '', getUrl({ ...diaryPage, contentEditable }));
```

Comment:
I dislike the spread operator in signatures: when changing the internal structure of `diaryPage`, the untyped interface of the function is getting jeopardized.
I prefer to pass the semantical object (`diaryPage`) and have the provider pick the necessary components using decomposition.

Reply:
Previously, you also gave the whole `diaryPage` into the `getUrl()` call with all it contains. The new call only adds the `contentEditable` attribute.
On the other hand, `getUrl()` will always only use the attributes which it requires, even if `diaryPage` is ever extended, and so stays compatible. In fact, it's an object that is given as a parameter, and the called function specifies, which attributes it requires, not the caller.

Reply:
ah, I saw the opening `{` in the wrong place.
Why did you opt for extending the object instead of creating a second parameter?

Reply:
I moved `contentEditable` out of the library function now as it doesn't have much to do with it.